pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/GodClassRule.java

```diff
@@ -4,366 +4,56 @@
-
-    /** The Weighted Method Count metric. */
-    private int wmcCounter;
-    /** The Access To Foreign Data metric. */
-    private int atfdCounter;
-
-    /**
-     * Collects for each method of the current class, which local attributes are
-     * accessed.
-     */
-    private Map<String, Set<String>> methodAttributeAccess;
-    /** The name of the current method. */
-    private String currentMethodName;
-
-    /**
-     * Base entry point for the visitor - the compilation unit (everything
-     * within one file). The metrics are initialized. Then the other nodes are
-     * visited. Afterwards the metrics are evaluated against fixed thresholds.
-     */
-    @Override
-    public Object visit(ASTCompilationUnit node, Object data) {
-        wmcCounter = 0;
-        atfdCounter = 0;
-        methodAttributeAccess = new HashMap<>();
-
-        Object result = super.visit(node, data);
-
-        double tcc = calculateTcc();
-
-        // StringBuilder debug = new StringBuilder();
-        // debug.append("Values for class ")
-        // .append(node.getImage()).append(": ")
-        // .append("WMC=").append(wmcCounter).append(", ")
-        // .append("ATFD=").append(atfdCounter).append(", ")
-        // .append("TCC=").append(tcc);
-        // System.out.println(debug.toString());
-
-        if (wmcCounter >= WMC_VERY_HIGH && atfdCounter > FEW_THRESHOLD && tcc < ONE_THIRD_THRESHOLD) {
-
-            StringBuilder sb = new StringBuilder();
-            sb.append(getMessage()).append(" (").append("WMC=").append(wmcCounter).append(", ").append("ATFD=")
-                    .append(atfdCounter).append(", ").append("TCC=").append(tcc).append(')');
-
-            RuleContext ctx = (RuleContext) data;
-            ctx.getReport().addRuleViolation(new JavaRuleViolation(this, ctx, node, sb.toString()));
-        }
-        return result;
-    }
-
-    /**
-     * Calculates the Tight Class Cohesion metric.
-     * 
-     * @return a value between 0 and 1.
-     */
-    private double calculateTcc() {
-        double tcc = 0.0;
-        int methodPairs = determineMethodPairs();
-        double totalMethodPairs = calculateTotalMethodPairs();
-        if (totalMethodPairs > 0) {
-            tcc = methodPairs / totalMethodPairs;
-        }
-        return tcc;
-    }
-
-    /**
-     * Calculates the number of possible method pairs. Its basically the sum of
-     * the first (methodCount - 1) integers. It will be 0, if no methods exist
-     * or only one method, means, if no pairs exist.
-     * 
-     * @return
+     * Few means between 2 and 5. See: Lanza. Object-Oriented Metrics in Practice. Page 18.
      */
-    private double calculateTotalMethodPairs() {
-        int methodCount = methodAttributeAccess.size();
-        int n = methodCount - 1;
-        double totalMethodPairs = n * (n + 1) / 2.0;
-        return totalMethodPairs;
-    }
+    private static final int FEW_ATFD_THRESHOLD = 6;
```

Comment:
Err no valid reason. I guess it's not uncommon to use getters or setters of the objects you manipulate, and a total of 5 in a class seemed a bit low (to me). I don't know if it makes a difference really, I can put back 5 so as not to change the behaviour too much